lib/perl/Genome/Model/Build/ClinSeq.pm

```diff
@@ -29,4 +29,59 @@ sub should_run_exome_cnv {
     return ($self->exome_build and $self->processing_profile->exome_cnv);
 }
 
+sub input_builds {
+    my $self = shift;
+
+    my @input_builds = ( $self->tumor_rnaseq_build, $self->normal_rnaseq_build );
+    for my $accessor (qw(wgs_build exome_build)) {
+        if ($self->$accessor) {
+            push @input_builds, $self->$accessor;
+            for my $build (qw(tumor_build normal_build)) {
+                if ($self->$accessor->can($build)) {
+                    push @input_builds, $self->$accessor->$build;
+                }
+            }
+        }
+    }
+
+    return grep {defined($_)} @input_builds;
+}
+
+sub reference_sequence_build {
+    my $self = shift;
+    return $self->model->_resolve_reference;
+}
+
+sub best_somatic_build {
+    my $self = shift;
+    if ($self->wgs_build) {
+        return $self->wgs_build;
+    }
+    elsif ($self->exome_build) {
+        return $self->exome_build;
+    }
+}
+
+sub best_somatic_build_subject_common_name {
+    my $self = shift;
+    return $self->best_somatic_build->subject->common_name;
+}
+
+sub has_microarray_build {
+    my $self = shift;
+
+    my $base_build;
+    if ($self->exome_build) {
+        $base_build = $self->exome_build;
+    }
+    elsif ($self->wgs_build) {
```

Comment:
Just a curiosity... but why doesn't `has_microarray_build` prefer the `best_somatic_build`?